cmd/scanners/aws/scanner.go

```diff
@@ -106,45 +110,55 @@ func startStockers() error {
 		if err != nil {
 			return err
 		}
-		// TODO handle error properly
 		inven.AddAccount(stockerInstance.GetResults())
 	}
 	return nil
 }
 
 func main() {
-	defer logger.Sync()
+	// Ignore Logger sync error
+	defer func() { _ = logger.Sync() }()
+
 	rdb, err := redis.InitDatabase(dbURL, dbPass)
 	if err != nil {
-		logger.Error("Failed to establish database connection", zap.Error(err))
+		logger.Fatal("Failed to establish database connection", zap.Error(err))
 	}
 	// Prepare New Stock
 	inven = inventory.NewInventory()
 
 	// Get Cloud Accounts from credentials file
-	accounts := GetCloudProviderAccounts()
+	accounts, err := getCloudProviderAccounts()
+	if err != nil {
+		logger.Error("Failed to get cloud provider accounts", zap.Error(err))
+		return
+	}
+
+	// Run Stockers
+	err = createStockers(accounts)
+
+	if err != nil {
+		logger.Error("Failed to add stockers", zap.Error(err))
+		return
+	}
 
-	// Running Stockers
-	// TODO Handle error properly
-	createStockers(accounts)
 	err = startStockers()
 	if err != nil {
-		logger.Fatal("Failed to start up stocker instances", zap.Error(err))
+		logger.Error("Failed to start up stocker instances", zap.Error(err))
 		return
 	}
 
 	b, err := json.Marshal(inven)
 	if err != nil {
-		logger.Fatal("Failed to marshal inventory data from DB", zap.Error(err))
+		logger.Error("Failed to marshal inventory data from database", zap.Error(err))
```

Comment:
Shouldn't be a "Fatal" to stop scanner execution?

Reply:
AFAIK `Fatal`` is a rare case
Here we return an error and exit the function.
No further code execution
For example, https://jhall.io/posts/2022-02-26-fatal-logging/

Reply:
Ok, got it